caps/types.go

```diff
@@ -65,3 +63,18 @@ func (t *Type) Validate(c *Capability) error {
 	}
 	return nil
 }
+
+// TypeLookupFn aids in looking up a Type by name
+type TypeLookupFn func(name string) *Type
+
+// MarshalJSON encodes a Type object as the name of the type.
+func (t *Type) MarshalJSON() ([]byte, error) {
+	return json.Marshal(t.Name)
```

Comment:
I think we want just the name as the type will have extra initialized, validated structures that we don't want to handle twice on the daemon side. Alternatively, we can hold those separate and just keep the type Type string; alias